base/applications/network/tracert/tracert.cpp

```diff
@@ -377,29 +377,26 @@ RunTraceRoute()
     }
 
     BYTE SendBuffer[PACKET_SIZE];
-    ICMPV6_ECHO_REPLY ReplyBufferv6;
-#ifdef _WIN64
-    ICMP_ECHO_REPLY32 ReplyBufferv432;
-#else
-    ICMP_ECHO_REPLY ReplyBufferv4;
-#endif
+
     PVOID ReplyBuffer;
 
     DWORD ReplySize = PACKET_SIZE + SIZEOF_ICMP_ERROR + SIZEOF_IO_STATUS_BLOCK;
     if (Info.Family == AF_INET6)
     {
-        ReplyBuffer = &ReplyBufferv6;
         ReplySize += sizeof(ICMPV6_ECHO_REPLY);
+
+        HANDLE heap = GetProcessHeap();
+        ReplyBuffer = HeapAlloc(heap, HEAP_ZERO_MEMORY, ReplySize);
     }
     else
     {
 #ifdef _WIN64
-        ReplyBuffer = &ReplyBufferv432;
         ReplySize += sizeof(ICMP_ECHO_REPLY32);
 #else
-        ReplyBuffer = &ReplyBufferv4;
         ReplySize += sizeof(ICMP_ECHO_REPLY);
 #endif
+        HANDLE heap = GetProcessHeap();
+        ReplyBuffer = HeapAlloc(heap, HEAP_ZERO_MEMORY, ReplySize);
```

Comment:
You can replace "heap" in the HeapAlloc() call directly by "GetProcessHeap()". Also you can factor out this heapallocation by moving it out of the if() and else() blocks.